### PR TITLE
applications: nrf_desktop: PAW3212 config

### PR DESCRIPTION
**Pull request by pdunaj**

### CODE REVIEW

**Comment by pdunaj** [comment]
Tested with red desktop mouse. Works fine.

**Comment by pdunaj** [comment]
Forgot to add header file to the commit. Included now and force pushed.

File: drivers/sensor/paw3212/paw3212.c [hunk 1]
```diff
@@ -257,6 +282,114 @@ static int reg_write(struct paw3212_data *dev_data, u8_t reg, u8_t val)
...
+	}
+
+	__ASSERT_NO_MSG((etm & PAW3212_ETM_MASK) == etm);
+
+	regval &= ~PAW3212_ETM_MASK;
+	regval |= etm;
+
+	err = reg_write(dev_data, reg_addr, regval);
+	if (err) {
+		LOG_ERR("Failed to change sleep time time");
```
Comment (MarekPieta) [inline drivers/sensor/paw3212/paw3212.c]: time time

File: include/sensor/paw3212.h [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+#ifndef ZEPHYR_INCLUDE_PAW3212_H_
+#define ZEPHYR_INCLUDE_PAW3212_H_
+
+#include <sensor.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum pmw3360_attribute {
```
Comment (MarekPieta) [inline include/sensor/paw3212.h]: paw3212

File: drivers/sensor/paw3212/paw3212.c [hunk 3]
```diff
@@ -257,6 +282,114 @@ static int reg_write(struct paw3212_data *dev_data, u8_t reg, u8_t val)
...
+		LOG_ERR("Failed to read sleep register");
+		return err;
+	}
+
+	__ASSERT_NO_MSG((etm & PAW3212_ETM_MASK) == etm);
+
+	regval &= ~PAW3212_ETM_MASK;
+	regval |= etm;
+
+	err = reg_write(dev_data, reg_addr, regval);
```
Comment (MarekPieta) [inline drivers/sensor/paw3212/paw3212.c]: Maybe try to re-enable write protect before returning error

File: drivers/sensor/paw3212/paw3212.c (hunk 1 above)
Comment (pdunaj) [reply drivers/sensor/paw3212/paw3212.c]: Fixed

File: include/sensor/paw3212.h (hunk 2 above)
Comment (pdunaj) [reply include/sensor/paw3212.h]: Fixed

File: drivers/sensor/paw3212/paw3212.c [hunk 4]
```diff
@@ -257,6 +282,114 @@ static int reg_write(struct paw3212_data *dev_data, u8_t reg, u8_t val)
...
+	err = reg_read(dev_data, reg_addr, &regval);
+	if (err) {
+		LOG_ERR("Failed to read sleep register");
+		return err;
+	}
+
+	__ASSERT_NO_MSG((etm & PAW3212_ETM_MASK) == etm);
+
+	regval &= ~PAW3212_ETM_MASK;
+	regval |= etm;
```
Comment (anangl) [inline drivers/sensor/paw3212/paw3212.c]: `(etm << PAW3212_ETM_POS)`, for consistency.

File: drivers/sensor/paw3212/paw3212.c [hunk 5]
```diff
@@ -634,6 +767,7 @@ static int paw3212_attr_set(struct device *dev, enum sensor_channel chan,
 			    const struct sensor_value *val)
 {
 	struct paw3212_data *dev_data = &paw3212_data;
+	int err;
```
Comment (anangl) [inline drivers/sensor/paw3212/paw3212.c]: I guess the function should return this (currently it returns 0 when the actual attribute change fails).

File: drivers/sensor/paw3212/paw3212.c (hunk 4 above)
Comment (pdunaj) [reply drivers/sensor/paw3212/paw3212.c]: Right! Fixed

File: drivers/sensor/paw3212/paw3212.c (hunk 5 above)
Comment (pdunaj) [reply drivers/sensor/paw3212/paw3212.c]: Thanks! fixed

**Pull request merged**